lib/std/zig/system/x86.zig

```diff
@@ -544,7 +544,7 @@ fn cpuid(leaf_id: u32, subid: u32) CpuidLeaf {
         : [leaf_id] "{eax}" (leaf_id),
           [subid] "{ecx}" (subid),
           [leaf_ptr] "r" (&cpuid_leaf),
-        : "eax", "ebx", "ecx", "edx"
+        : "ebx", "edx" // "eax" and "ecx" are already inputs
```

Comment:
Hmm. I'm pretty sure these need to be marked as outputs, otherwise it will be assumed that they are unmodified. leaf_ptr might also need to be `=*m`, so that its memory is known to be clobbered

Zig does support multiple asm outputs now though, so maybe this is simpler:
```zig
asm volatile (
    \\ cpuid
    : [_] "={eax}" (eax),
      [_] "={ebx}" (ebx),
      [_] "={ecx}" (ecx),
      [_] "={edx}" (edx),
    : [_] "0" (leaf_id),
      [_] "2" (subid),
);
```

Reply:
> Hmm. I'm pretty sure these need to be marked as outputs, otherwise it will be assumed that they are unmodified.

This does appear to be correct, but additionally the behavior on gcc is so much worse in that even adding an output does not help: [link](https://godbolt.org/z/osc8GxGY1).  It's a good thing we have multiple outputs, which should work much better.